src/Client/DisqusClient.php

```diff
@@ -95,66 +86,40 @@ public function fetch(string $shortname, array $options, string $fetch = 'thread
         return $content;
     }
 
-    public function getParameters(): array
-    {
-        return [
-            'id' => $this->id,
-            'debug' => $this->debug,
-            'api_key' => $this->apiKey,
-        ];
-    }
-
-    public function getSsoParameters(array $parameters): array
-    {
-        $sso = [];
-
-        if ($this->secretKey && isset($parameters['sso']) && isset($parameters['sso']['user'])) {
-            $sso = $parameters['sso'];
-
-            if (isset($sso['user'])) {
-                $message = base64_encode(json_encode($sso['user']));
-                $timestamp = time();
-                $hmac = hash_hmac('sha1', "$message $timestamp", $this->secretKey);
-
-                unset($sso['user']);
-                $sso['auth'] = [
-                    'message' => $message,
-                    'hmac' => $hmac,
-                    'timestamp' => $timestamp,
-                ];
-            }
-        }
-
-        return $sso;
-    }
-
     private function buildUrl(string $shortname, array $options, string $fetch, string $format = 'json'): string
     {
         if (isset($options['identifier'])) {
-            $this->id = [
+            $this->config->setId([
```

Comment:
This is something we should finally fix (while we're breaking things). The flow is:

A) `DisqusHelper` calls `DisqusClient::fetch()`
B) `DisqusClient::fetch()` calls `DisqusClient::buildUrl()`, which *mutates* the `DisqusConfig` object. This is the "bad behavior" we don't want - it's unnatural (but it was always this way).
C) After `DisqusHelper` calls `DisqusClient::fetch()` (step A), it calls `DisqusConfig::getParameters()`, which uses the mutation that occurred in B.

Here's what we should do instead:

1) Don't modify the DisqusConfig in this method. Probably, in this method (`buildUrl()`) we will need to call some new method on `DisqusConfig` to get the proper value for the `$id` variable (basically, move all these "ifs" into some new method like `getIdentifierQueryParam()` in `DisqusConfig` but do NOT mutate `DisqusConfig` in that method
2) In `DisqusHelper`, when we call `DisqusConfig::getParameters()`, let's now pass in the `$options` array. So, `DisqusConfig::getParameters()` will have an `array $options` argument. It can then use that array to determine the the identifier param (e.g. `identifier`, `link` or `id`) to return.